foundation/src/commonMain/kotlin/com/mirego/trikot/foundation/ref/WeakProperty.kt

```diff
@@ -0,0 +1,35 @@
+package com.mirego.trikot.foundation.ref
+
+import com.mirego.trikot.foundation.concurrent.AtomicReference
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+/**
+ * Creates weak atomic reference property delegate.
+ * It can only be used in initialize of read-write optional property, like this:
+ *
+ * ```
+ * val weakType : Type? by weakReference()
+ * ```
+ */
+fun <T : Any> weakAtomicReference() =
+    WeakProperty<T>()
+
+/**
+ * Weak atomic reference to a variable of type [T].
+ */
+class WeakProperty<T : Any> : ReadWriteProperty<Any?, T?> {
+    private val internalNavigationDelegateRef = AtomicReference<WeakReference<T>?>(null)
```

Comment:
I would rename to `internalReference`

Reply:
That is a great idea, I will push a ninja fix.